### PR TITLE
API Rework

### PR DESCRIPTION
**Pull request by petermorgandev**
This took me a while, but I managed to successfully refactor the API. I reduced the lines of code from 421 to 310, but more importantly I made it easier to understand what the code actually does.

I am not sure what the purpose of the `routes.get('/*', (request, result) =>` route is. I would be happy to work on refactoring it given some guidance on what it does.

In testing the refactoring I made this change to routes/index.js:
```javascript
// const apiV1 = require('./api/v1');
// routes.use('/api', apiV1);
const apiV2 = require('./api/v2');
routes.use('/api', apiV2);
```

### CODE REVIEW

**Comment by jrtechs** [comment]
@PeterMorganGH the * route is currently being used for pulling general information on organizations or users. This information is used in the profile view box. This will ideally get refactored out shortly because like the friends and repositories queries, this has a bunch of unused JSON properties.

**Comment by petermorgandev** [comment]
@jrtechs I see now. Thanks for the explanation.

File: routes/api/v2.js [hunk 1]
```diff
@@ -0,0 +1,310 @@
+const routes = require('express').Router();
```
Comment (jrtechs) [inline routes/api/v2.js:1]: Moving forward we should start adding top level descriptions for files

File: routes/api/v2.js [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+routes.get("/friends/:name", async (req, res)=> {
+    try {
+        const query = await queryFriends(req.params.name);
+        res.json(query);
+    } catch (error) {
+        res.status(500).json({error: 'API error fetching friends'});
+    }
+});
+
+
```
Comment (jrtechs) [inline routes/api/v2.js:236]: hmm, we should make some API docs somewhere

**Review by jrtechs — APPROVED** [review]
@PeterMorganGH changes look great! In the future if this file gets any bigger we might want to refactor it out into a few different files.

File: routes/api/v2.js (hunk 2 above)
Comment (petermorgandev) [reply routes/api/v2.js:236]: I can work on starting that effort today.

Comment (jrtechs) [reply routes/api/v2.js:236]: https://github.com/jrtechs/github-graphs/issues/29

**Pull request merged**